src/EditorFeatures/CSharpTest2/Recommendations/WhenKeywordRecommenderTests.cs

```diff
@@ -193,33 +193,33 @@ public async Task TestForCatchClause_NotAfterFilter3()
         public async Task TestForSwitchCase_NotInEmptySwitchStatement() =>
             await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { $$ }"));
 
-        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/40015"), Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
         public async Task TestForSwitchCase_SemanticCheck_NotAfterPredefinedType() =>
-            await VerifyAbsenceAsync(AddInsideMethod(@"switch (new object()) { case int $$ }"));
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (new object()) { case int $$ }"));
```

Comment:
this is the newly allowed behavior.  i.e. you can say `case int when ...` instead of `case int i when ...`